src/main/java/javax/validation/ConstraintViolationException.java

```diff
@@ -32,7 +34,7 @@ public ConstraintViolationException(String message,
 			this.constraintViolations = null;
 		}
 		else {
-			this.constraintViolations = new HashSet<ConstraintViolation<?>>( constraintViolations );
+			this.constraintViolations = new HashSet<>( constraintViolations );
```

Comment:
I had been thinking about this, too. I refrained from doing it as it's behavioral change that's not strictly needed. But then the spec makes no statement on the mutability of this set, so we should be fine to change it.